### PR TITLE
Added History section in the application

### PR DESCRIPTION
**Pull request by H-SM**
using the mongo backend and extracting the history for that user... currently only the state is connected up with the history section.. we can't add new details in it for now as that gateway still need to be build up.. **right now its just how it will finally look over the UI** the favourite option and the view option hasn't been linked to there api calls yet. 

For now it's extracting the user's info and showcasing all the details over the table field... 

image

### CODE REVIEW

File: client/src/components/History.js [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+                          scope="col"
+                          class="py-[0.875rem] pl-[1rem] pr-[0.75rem] text-left text-[0.875rem] leading-5 font-bold text-opacity-100 text-gray-700 px-6"
+                        >
+                          Edges
+                        </th>
+                        <th
+                          scope="col"
+                          class="py-[0.875rem] pl-[1rem] pr-[0.75rem] text-left text-[0.875rem] leading-5 font-bold text-opacity-100 text-gray-700 px-6"
+                        >
+                          Time Complexity
```
Comment (Sid-the-Slothh) [inline client/src/components/History.js:132]: Just "time taken" would be better.

File: client/src/components/History.js [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+                        </th>
+                        <th
+                          scope="col"
+                          class="py-[0.875rem] pl-[1rem] pr-[0.75rem] text-left text-[0.875rem] leading-5 font-bold text-opacity-100 text-gray-700 px-6"
+                        >
+                          Space Complexity
```
Comment (Sid-the-Slothh) [inline client/src/components/History.js:138]: And just "space taken" would also be more fitting here.

File: client/src/components/History.js [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+                                    →
+                                  </span>
+                                </a>
+                              </td>
+                            </tr>
+                          );
+                        })
+                      ) : (
+                        <>
+                            {/* will have the search logic here */}
```
Comment (Sid-the-Slothh) [inline client/src/components/History.js:243]: Put a TODO here maybe idk :)

File: server/routes/graph.js [hunk 4]
```diff
@@ -19,18 +19,26 @@ router.get("/fetchallgraphs", fetchuser, async (req, res) => {
...
+    const { result, graph, name, favourite } = req.body;
     const errors = validationResult(req);
     if (!errors.isEmpty()) {
       return res.status(400).json({ errors: errors.array() });
     }
-    
+    //look over unique naming
+    const existingGraph = await Graph.findOne({ name, user: req.user.id });
+    if (existingGraph) {
+      return res.status(400).json({ status: "validation error", errors: "Name must be unique for the user" });
```
Comment (Sid-the-Slothh) [inline server/routes/graph.js:35]: This error message is a bit unclear. "Graph name must be ..." would be better.

Comment (Sid-the-Slothh) [reply server/routes/graph.js:35]: Ok if its the backend then WHAtever english be damned amirite

**Review by Sid-the-Slothh — APPROVED** [review]
ok

**Review by ThapaVinay — APPROVED** [review]

**Pull request merged**